src/test/scala/com/fulcrumgenomics/vcf/ByIntervalListVariantContextIteratorTest.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.vcf
+
+import com.fulcrumgenomics.testing.{UnitSpec, VariantContextSetBuilder}
+import htsjdk.samtools.SAMFileHeader
+import htsjdk.samtools.util.{Interval, IntervalList}
+
+/**
+  * Tests for ByIntervalListVariantContextIterator.
+  */
+class ByIntervalListVariantContextIteratorTest extends UnitSpec {
+
+  private val dict = new VariantContextSetBuilder().header.getSequenceDictionary
+
+  private def emtpyIntervalList(): IntervalList = {
+    val header = new SAMFileHeader
+    header.setSequenceDictionary(this.dict)
+    new IntervalList(header)
+  }
+
+  "ByIntervalListVariantContextIterator" should "return no variants if the interval list is empty" in {
```

Comment:
A couple more tests I would think to add:
1. How about Indels that span the start or end of an interval (i.e. a deletion from 495-505 and an interval from 500-1000)
2. I think that the using-index mechanism will return the same variant twice if a variant overlaps two intervals that can't be merged because they are more than 1bp apart:
    Deletion spanning 495-505
    Interval1 is 400-498
    Interval2 is 502-599